components/codeflare/codeflare.go

```diff
@@ -1,140 +0,0 @@
-		log.Error(err, "failed to update image", "path", CodeflarePath+"/bases")
-	}
-
-	return nil
-}
-
-func (c *CodeFlare) OverrideManifests(ctx context.Context, _ cluster.Platform) error {
-	// If devflags are set, update default manifests path
-	if len(c.DevFlags.Manifests) != 0 {
-		manifestConfig := c.DevFlags.Manifests[0]
-		if err := deploy.DownloadManifests(ctx, ComponentName, manifestConfig); err != nil {
-			return err
-		}
-		// If overlay is defined, update paths
-		defaultKustomizePath := "default"
-		if manifestConfig.SourcePath != "" {
-			defaultKustomizePath = manifestConfig.SourcePath
-		}
-		CodeflarePath = filepath.Join(deploy.DefaultManifestPath, ComponentName, defaultKustomizePath)
-	}
-
-	return nil
-}
-
-func (c *CodeFlare) GetComponentName() string {
-	return ComponentName
-}
-
-func (c *CodeFlare) ReconcileComponent(ctx context.Context,
-	cli client.Client,
-	l logr.Logger,
-	owner metav1.Object,
-	dscispec *dsciv1.DSCInitializationSpec,
-	platform cluster.Platform,
-	_ bool) error {
-	enabled := c.GetManagementState() == operatorv1.Managed
-	monitoringEnabled := dscispec.Monitoring.ManagementState == operatorv1.Managed
-
-	if enabled {
-		if c.DevFlags != nil {
-			// Download manifests and update paths
-			if err := c.OverrideManifests(ctx, platform); err != nil {
-				return err
-			}
-		}
-		// check if the CodeFlare operator is installed: it should not be installed
-		// Both ODH and RHOAI should have the same operator name
-		dependentOperator := CodeflareOperator
-
-		if found, err := cluster.OperatorExists(ctx, cli, dependentOperator); err != nil {
-			return fmt.Errorf("operator exists throws error %w", err)
-		} else if found {
-			return fmt.Errorf("operator %s is found. Please uninstall the operator before enabling %s component",
-				dependentOperator, ComponentName)
-		}
-
-		// It updates stock manifests, overridden manifests should contain proper namespace
-		if err := deploy.ApplyParams(ParamsPath, nil, map[string]string{"namespace": dscispec.ApplicationsNamespace}); err != nil {
-			return fmt.Errorf("failed update image from %s : %w", CodeflarePath+"/bases", err)
-		}
-	}
-
-	// Deploy Codeflare
-	if err := deploy.DeployManifestsFromPath(ctx, cli, owner, //nolint:revive,nolintlint
-		CodeflarePath,
-		dscispec.ApplicationsNamespace,
-		ComponentName, enabled); err != nil {
-		return err
-	}
-	l.Info("apply manifests done")
-
-	if enabled {
-		if err := cluster.WaitForDeploymentAvailable(ctx, cli, ComponentName, dscispec.ApplicationsNamespace, 20, 2); err != nil {
-			return fmt.Errorf("deployment for %s is not ready to server: %w", ComponentName, err)
-		}
-	}
-
-	// CloudServiceMonitoring handling
-	if platform == cluster.ManagedRhods {
-		// inject prometheus codeflare*.rules in to /opt/manifests/monitoring/prometheus/prometheus-configs.yaml
-		if err := c.UpdatePrometheusConfig(cli, l, enabled && monitoringEnabled, ComponentName); err != nil {
```

Comment:
@lburgazzoli Just wanted to ask, I have removed the `PrometheusRule` from the codeflare-controller but I'm guessing this functionality get impacted with this removal?

Reply:
we will take the monitoring part into "service" deal with that in different PR